src/main/scala/net/snowflake/spark/snowflake/pushdowns/querygeneration/WindowStatement.scala

```diff
@@ -26,11 +31,70 @@ private[querygeneration] object WindowStatement {
     expAttr: (Expression, Seq[Attribute])
   ): Option[SnowflakeSQLStatement] = {
     val expr = expAttr._1
+    val fields = expAttr._2
 
     Option(expr match {
-      case _: RowNumber => ConstantString(expr.prettyName.toUpperCase) + "()"
+      // Handle Window Expression. (It is moved from MiscStatement)
+      case WindowExpression(func, spec) =>
+        func match {
+          // These functions in Snowflake support a window frame.
+          // Note that pushdown for these may or may not yet be supported in the connector.
+          case _: Rank | _: DenseRank | _: PercentRank =>
+            convertStatement(func, fields) + " OVER " + windowBlock(
+              spec,
+              fields,
+              useWindowFrame = true
+            )
+
+          // These do not.
+          case _ =>
```

Comment:
Do these functions work correctly without the window frame? I mean perform same as Spark functions?

Reply:
I don't think so. As the document, rank() and dense_rank() are only included in  Window functions part in Spark document. I believe they are window functions only.
Reference to https://spark.apache.org/docs/latest/sql-ref-functions-builtin.html#window-functions

Reply:
I mean, in this `case _`. you set useWindowFrame to false. Do you have any example of window function works without window frame? Does this case always fail?